### ISSUE TITLE
snapshotController.installDefaultClass = false does not work

### ISSUE THREAD

**Issue by yurrriq** [kubernetes/kops#12244]
/kind bug

**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Version 1.21.1 (git-1.21.1)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

- `kops edit cluster`
- set `spec.snapshotController.installDefaultClass = false`

**5. What happened after the commands executed?**

```
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# Found fields that are not recognized
# ...
#     snapshotController:
#       enabled: true
# +     installDefaultClass: false
```

**6. What did you expect to happen?**

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```yaml
spec:
  snapshotController:
    enabled: true
    installDefaultClass: false
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

Can do later, if needed.

**9. Anything else do we need to know?**

I wonder if the `bool` needs to be a `*bool` like the others.

**Comment by olemarkus** [comment]
The way kOps decides a flag doesn't exist is by demarshaling/marshaling and then diff. Since `false` will be omitted, kOps thinks the field doesn't exist. It will be some work to fix how kOps determines what fields exists or not.

For now, just don't set the field as doing so is noop anyway.

**Comment by olemarkus** [comment]
This should be fixed by #12804

/remove-lifecycle stale
